### PR TITLE
Chore/refactaroo for Ops in synapse/workload

### PR DESCRIPTION
**Pull request by merkata**
### Overview

Refactors some bits from workload to not rely on `ops` - currently the charm passes a container to src/pebble which then passes it to src/synapse/workload and the workload makes a lot of (duplicate) things that are perhaps better handled at one layer and not across. 

Note that this is against the irc bridge branch, as the proposal to refactor came in there and if this gets accepted it will land in the PR for the irc bridge.

### Rationale

Reduce code and complexity, make dev person happi.

### Juju Events Changes

n/a

### Module Changes

moved Ops pieces (checks and config manipulation) to pebble, still WIP

### Library Changes

n/a

### Checklist

- [x ] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x ] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x ] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x ] The documentation is generated using `src-docs`
- [x ] The documentation for charmhub is updated.
- [x ] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/pebble.py [hunk 1]
```diff
@@ -121,36 +224,36 @@ def change_config(charm_state: CharmState, container: ops.model.Container) -> No
     """
     try:
         synapse.execute_migrate_config(container=container, charm_state=charm_state)
-        synapse.enable_metrics(container=container)
-        synapse.enable_forgotten_room_retention(container=container)
-        synapse.enable_serve_server_wellknown(container=container)
+        current_yaml = _get_synapse_config(container)
```
Comment (nrobinaubertin) [inline src/pebble.py:227]: I wonder if `current_yaml` is a good name. Maybe `current_synapse_config` ?

File: src/pebble.py [hunk 2]
```diff
@@ -121,36 +224,36 @@ def change_config(charm_state: CharmState, container: ops.model.Container) -> No
...
-            synapse.enable_trusted_key_servers(container=container, charm_state=charm_state)
+            synapse.enable_trusted_key_servers(current_yaml, charm_state=charm_state)
         if charm_state.synapse_config.ip_range_whitelist:
-            synapse.enable_ip_range_whitelist(container=container, charm_state=charm_state)
+            synapse.enable_ip_range_whitelist(current_yaml, charm_state=charm_state)
         if charm_state.datasource:
             logger.info("Synapse Stats Exporter enabled.")
             replan_stats_exporter(container=container, charm_state=charm_state)
-        synapse.validate_config(container=container)
+        synapse.validate_config(container)
```
Comment (amandahla) [inline src/pebble.py:255]: The yaml should be updated in the container before running validate_config

Comment (amandahla) [reply src/pebble.py:255]: Or could be refactored to run in a temporary file, it would work too.

**Comment by amandahla** [comment]
Nice refactor, thanks!

File: src/pebble.py (hunk 2 above)
Comment (merkata) [reply src/pebble.py:260]: Good catch! I moved it after pushing (for now).

File: src/pebble.py (hunk 1 above)
Comment (merkata) [reply src/pebble.py]: It is better, renamed. Thanks!

**Pull request merged**